### PR TITLE
[usage] Setup usage reset job

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Initial setup for usage reset job. Currently does nothing but logs when it's triggered. 
Config in the installer will be added in subsequent PR.

## Related Issue(s)

* Part of https://github.com/gitpod-io/gitpod/issues/14178

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

File: components/usage/pkg/scheduler/reset_usage_job.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+package scheduler
+
+import (
+	"time"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+)
+
+func NewResetUsageJobSpec(schedule time.Duration) (JobSpec, error) {
```
Comment (andrew-farries) [inline components/usage/pkg/scheduler/reset_usage_job.go:13]: The constructor is called `NewResetUsageJobSpec` but the type is simply `ResetUsageJob`. Do you want to make them consistent?

File: components/usage/pkg/server/server.go [hunk 2]
```diff
@@ -138,6 +141,20 @@ func Start(cfg Config, version string) error {
 		log.Info("No controller schedule specified, controller will be disabled.")
 	}
 
+	if cfg.ResetUsageSchedule != "" {
+		schedule, err := time.ParseDuration(cfg.ResetUsageSchedule)
+		if err != nil {
+			return fmt.Errorf("failed to parse reset usage schedule as duration: %w", err)
+		}
+
+		jobSepc, err := scheduler.NewResetUsageJobSpec(schedule)
```
Comment (andrew-farries) [inline components/usage/pkg/server/server.go:150]: typo here, `jobSepc`

**Review by andrew-farries — CHANGES_REQUESTED** [review]

File: components/usage/pkg/scheduler/reset_usage_job.go (hunk 1 above)
Comment (easyCZ) [reply components/usage/pkg/scheduler/reset_usage_job.go:13]: done

File: components/usage/pkg/server/server.go (hunk 2 above)
Comment (easyCZ) [reply components/usage/pkg/server/server.go]: fixed, thanks.

**Comment by svenefftinge** [comment]
I would have preferred to call this in the existing ledger job after the sync. I think it's simpler and would also help with making sure the usage for the time we use was reconciled.

**Comment by easyCZ** [comment]
@svenefftinge My thinking here is that the usage reset should be lighter weight in terms of compute time and should run more frequently. I wanted to avoid coupling them if we need to adjust the frequency on either without being limited to the runtime of the usage reconcile.

But you make a good point. They could run in the same control loop. In either case, we shouldn't assume that the reset runs shortly after usage reconcile as that may not hold true.

**Pull request merged**